src/cpufreq.c

```diff
@@ -39,19 +44,44 @@ struct feat_t {
     bool pcluster_only;
 };
 
+static u32 pstate_reg_to_pstate(u64 val)
+{
+    switch (chip_id) {
+        case S5L8960X:
+        case T7000 ... T7001:
+            return FIELD_GET(CLUSTER_PSTATE_DESIRED1_S5L8960X, val);
+        case S8000 ... T8122:
+        case T6000 ... T6034:
+        default:
+            return FIELD_GET(CLUSTER_PSTATE_DESIRED1, val);
+    }
+}
+
 static int set_pstate(const struct cluster_t *cluster, uint32_t pstate)
 {
     u64 val = read64(cluster->base + CLUSTER_PSTATE);
 
-    if (FIELD_GET(CLUSTER_PSTATE_DESIRED1, val) != pstate) {
-        val &= ~CLUSTER_PSTATE_DESIRED1;
-        val |= CLUSTER_PSTATE_SET | FIELD_PREP(CLUSTER_PSTATE_DESIRED1, pstate);
-        if (chip_id == T8103 || chip_id <= T6002) {
-            val &= ~CLUSTER_PSTATE_DESIRED2;
-            val |= CLUSTER_PSTATE_SET | FIELD_PREP(CLUSTER_PSTATE_DESIRED2, pstate);
+    if (pstate_reg_to_pstate(val) != pstate) {
+        switch (chip_id) {
+            case S5L8960X:
+            case T7000 ... T7001:
```

Comment:
This is now just hiding the comparisons that I'd like to avoid. Please get rid of the ... stuff as well and just open code all the chip_ids here.